tasks/workflow/task_execution.go

```diff
@@ -180,8 +181,18 @@ func setTaskStatus(kv *api.KV, taskID string, status tasks.TaskStatus, lastIndex
 	}
 	if !set {
 		log.Debugf("[WARNING] Failed to set task status to:%q for taskID:%q as last index has been changed before. Retry it", status.String(), taskID)
-		return checkAndSetTaskStatus(kv, taskID, status)
+		return checkAndSetTaskStatus(ctx, kv, targetID, taskID, status)
 	}
+
+	// Emit event for status change
+	// As task has been set, error are ignored
+	wfName, _ := tasks.GetTaskData(kv, taskID, "workflowName")
```

Comment:
What will happen if wfName is empty (in case of error silently bypassed)?

Reply:
wfName may be empty as this data is not filled for non-workflow task type (as for custom command by instance)

Reply:
I add some comments

Reply:
Sounds good.